satellite/admin-back-office-v1.md

```diff
@@ -0,0 +1,646 @@
+- Storage limit, and select GB or TB.
+- Egress limit, and select GB or TB
+- Segment limit, and enter the max number with formatted numerical input fields for better
+  readability.
+
+__When an account needs to be deleted__
+
+I want authorized users to be able to press a "delete account" button along with a "confirm" delete
+button that confirms this user and all their information and stored data are to be deleted, so that
+we don't have to manually access their API keys and manually delete their data.
+
+__When an account needs to be created__
+
+I want authorized users to have an option in their initial login view to create a user.
+
+This operation rarely happens so the design of the UX shouldn’t make it prominent
+
+Acceptance criteria:
+- Input fields for: full name, email, generate (temporary) password.
+
+__When an account needs more projects__
+
+I want to locate the user I need to enable a higher project allowance for and have an option to
+increase the number of projects the account is allowed to have in the account details view.
+
+Acceptance criteria:
+- Field to increase or decrease the account's project count.
+- Project limits default to the user's project limits or can be set if otherwise.
+
+__When a user needs to be added to a different user's project__
+
+I want to be able to search for the project ID.
+I need to add the user to, so that I can view the members off that project and add and/or remove a
+user.
+
+__When a user needs their multi factor authentication (MFA) reset__
+
+I want there to be a MFA reset option in the account details view.
+
+Acceptance criteria:
+- We can see that MFA is enabled or disabled.
+- We can only disable MFA.
+
+__When a user wants to change the email address tied to their account__
+
+I want to be able to edit the email address in the account details view so that all events and data
+associated to that user is now under the updated email address.
+
+##### Modifications history
+
+The back office keeps a register of all performed operations and who executed them that cause
+modifications.
+
+Each record has the following information:
+- Timestamp. When the operation was performed.
+- Operator email. Who performed the operation.
+- Account ID. What account was modified.
+- Entity that was modified (i.e. account, project, bucket) and ID.
+- Operation. What was done (e.g. change email address, set user agent, etc.). The text used to
+  identify the operation has to be unique inside of the entity and human friendly.
+- Current data. The new data that was modified with this operation. This is a structured blob
+  because it could be more than one modified field per operation.
+- Previous data. The existing data previous to the modification performed by this operation. This is
+  a structured blob because it could be more than one modified field per operation.
+
+This information is stored in the same database that the satellite uses.
```

Comment:
The big question is why do we keep a record in the first place. Do we want to have a full audit log in case someone abuses the admin interface? If so we should not store the log in a location that the same person can most likely modify.

Let's say I want to leave the company and blow up everything on my way out. I would start to modify a bunch of users, projects and buckets and for each modification I make sure to delete the evidence from the database. At that point the log is kind of useless to catch me unless you protect the log in a way that I don't have permission to alter it.

Reply:
I think that is a valid point, and it would be true in an extreme case. Still, I think having an audit log is useful regardless:
* perhaps someone who wants to blow up something will be deterred just by the fact of there being logs
* perhaps someone who wants to blow up something does not have `delete` abilities on the audit table

But personally, I think it is a great idea for one reason, completely not related to breaking things/malicious employees:

It gives us a very easy way to see if anything could have been caused by an admin if diagnosing issues for specific users. Let's say someone submits a support ticket due to an issue, or a customer complains about something to the sales team. Whoever is investigating would probably be interested in any recent admin actions on that customer, and why they were made. If there is a table, we can identify these actions as specific events that occurred at a point in time, which (imo) helps a lot for investigation work.

Reply:
The modification history is more for what Moby said.
Moreover, we are showing it in the same web UI.

Reply:
About "showing the history in the Web UI" you can see in the already-built frontend https://storj.github.io/admin-ui  For example, navigate to see a user details page and at the bottom of the page

![image](https://github.com/storj/design-docs/assets/1731633/556f6721-6632-4fc0-a35d-22c76ebb45b1)

Reply:
I think we can figure out that detail later. For now good to merge from my side. Adding a text field is nothing that is worth blocking the PR.

Reply:
That's a good idea.
@mobyvb  thought something similar, but more generic https://github.com/storj/design-docs/pull/1#discussion_r1296310794

We could add what Moby said, and then the UI could lead the operator to set specific information when performing an action. I know that people could cheat by making an API call and bypassing UI, but this is for internal use, and I like to trust who is working at Storj, considering that some employee does it, we could find who did it and take whatever measures.

Anyway, as I commented to Moby, that's something that we can add later without that much hassle, and if we want it for this first initial version then:
a) If we add it without specifying what, it seems to add "premature features".
b) If we specify it now, we will delay the design doc for a while more, because we should have some meetings with product owners and Tome, to modify the use cases and the UI. 

Doing a) is bad, for the reasons that I gave to Moby.
Doing b) isn't worth it, for something that we can add without that much trouble in some of the next versions.

Nonetheless, I'm going to add this to the out-of-scope section as a reminder of this conversation and a feature that we may want/need.